ivy/utils/backend/handler.py

```diff
@@ -252,11 +252,13 @@ def _set_backend_as_ivy(
     backend_str = backend.current_backend_str() if backend_str is None else backend_str
     for k, v in original_dict.items():
         compositional = k not in backend.__dict__
-        if k not in backend.__dict__:
+        if compositional:
             if k in invalid_dtypes and k in target.__dict__:
                 del target.__dict__[k]
                 continue
+            v = _verify_efficient_implementations(v) if v else v
             backend.__dict__[k] = v
+        v = _verify_efficient_implementations(v) if v else v
```

Comment:
probably we could do this before the `if compositional` block to avoid duplication

Reply:
yeah, right